src/System.Data.SqlClient/src/System/Data/SqlClient/SqlCommand.cs

```diff
@@ -290,7 +293,8 @@ internal SqlStatistics Statistics
             {
                 if (null != _activeConnection)
                 {
-                    if (_activeConnection.StatisticsEnabled)
+                    if (_activeConnection.StatisticsEnabled || 
+                        _diagnosticListener.IsEnabled(SqlClientDiagnosticListenerExtensions.SqlAfterExecuteCommand))
```

Comment:
@nbilling How is the observer registered? Any samples on the usage of this capability? 
What code / class should I look at, to see what the observer is going to look like?